### PR TITLE
[test][eks] fix mpi operator

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
The latest release for MPI operator [v0.3.0](https://github.com/kubeflow/mpi-operator/releases/tag/v0.3.0) introduced a breaking change, hence upgrading the version to sync up the changes.

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dlc_developer_config.toml [hunk 1]
```diff
@@ -16,19 +16,19 @@ benchmark_mode = false
 
 [build]
 # Frameworks for which you want to disable both builds and tests
-skip_frameworks = []
+skip_frameworks = ["pytorch", "huggingface_tensorflow", "huggingface_pytorch"]
```
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:19]: To be reverted before PR merge

File: test/dlc_tests/eks/mxnet/training/test_eks_mxnet_multinode_training.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 LOGGER = eks_utils.LOGGER
 
 
-@pytest.mark.skipif(is_pr_context(), reason=SKIP_PR_REASON)
+#@pytest.mark.skipif(is_pr_context(), reason=SKIP_PR_REASON)
```
Comment (tejaschumbalkar) [inline test/dlc_tests/eks/mxnet/training/test_eks_mxnet_multinode_training.py:20]: To be reverted before PR merge

File: test/dlc_tests/eks/tensorflow/training/test_eks_tensorflow_multi_node_training.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 
 # Test only runs in region us-west-2, on instance type p3.16xlarge, on PR_EKS_CLUSTER_NAME_TEMPLATE cluster
-@pytest.mark.skipif(is_pr_context(), reason=SKIP_PR_REASON)
+#@pytest.mark.skipif(is_pr_context(), reason=SKIP_PR_REASON)
```
Comment (tejaschumbalkar) [inline test/dlc_tests/eks/tensorflow/training/test_eks_tensorflow_multi_node_training.py:17]: To be reverted before PR merge

File: test/dlc_tests/eks/mxnet/training/test_eks_mxnet_multinode_training.py (hunk 2 above)
Comment (arjkesh) [reply test/dlc_tests/eks/mxnet/training/test_eks_mxnet_multinode_training.py:20]: probably we could add an eks multinode option in the developer config file. Out of scope for this PR

Comment (shantanutrip) [reply test/dlc_tests/eks/mxnet/training/test_eks_mxnet_multinode_training.py:20]: Surely, we should do that. Can we make that a future task? Just wanting to avoid delays for now.

**Review by shantanutrip — APPROVED** [review]

File: test/dlc_tests/eks/mxnet/training/test_eks_mxnet_multinode_training.py (hunk 2 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/eks/mxnet/training/test_eks_mxnet_multinode_training.py]: EKS test infra is not designed to run on PR context. We might have to add extra configuration other than dev config file

**Review by shantanutrip — APPROVED** [review]
